addons/mrp/models/stock_move.py

```diff
@@ -291,6 +291,17 @@ def create(self, vals_list):
         return super().create(vals_list)
 
     def write(self, vals):
+        is_product_id_updated = 'product_id' in vals and self.product_id.id != vals.get('product_id')
```

Comment:
self.product_id may be a multi recordset so `.id` may trigger a traceback

Reply:
Hello @Whenrow , 
Thanks for your suggestion. 
In this particular by-product update case, there is only a single record in self, but there may be a few cases where multiple recordsets are involved, which I missed. This could cause a traceback error.

Now I have updated the code.
Please check.
Thanks!

Reply:
Hello @Whenrow 
Reminder! 
Thanks!